### DISCUSSION TITLE
Error when trying to run migrations on Linux

### DISCUSSION THREAD

**Discussion by solucaodev** [adonisjs/core#3576] [Help]
When trying to "run migration:run" using NodeJS on Linux, the framework is unable to check which migrations have already been applied before, so it tries to apply everything again.

When running "migration:status", I get a table with all migrations with "corrupt" status.

I didn't find any documentation about the problem.

Grateful for the attention!

**Comment by guntribam** [comment]
please share more details: env variables, db used, logs, etc

**Reply by solucaodev** [reply]
Não há nada muito específico no .env, mas segue:
`PORT=3333
HOST=0.0.0.0
NODE_ENV=development
APP_KEY=XIv33***********izcuhExXnMkme
DRIVE_DISK=local
DB_CONNECTION=mssql
MSSQL_SERVER=****
MSSQL_PORT=1433
MSSQL_USER=****
MSSQL_PASSWORD=*****
MSSQL_DB_NAME=****`

We are using SQL Server.

**It is important to note that on Windows it works. Only when I try to run via Linux does it not work.**

I will recreate the environment in a virtual machine. I went back to Windows to be able to work.

**Reply by guntribam** [reply]
i use linux and mssql ... and it runs without any problems. I don't know how to help you with the info you've provided. What i can say it is to try to run a simple query on an endpoint like:
`await Database.rawQuery('select * from TESTE')`

check to see if connects

my mssql run on docker, and i've used this to start:
```
docker run --name=mssql --env=SA_PASSWORD=myStrong(!)Password --env=ACCEPT_EULA=Y -p 1433:1433 --restart always mcr.microsoft.com/mssql/server:latest
```

**Reply by guntribam** [reply]
This is my connection:

```
app: {
      client: 'mssql',
      connection: {
        ...(Env.get("MSSQL_DOMAIN") && {type: "ntlm", domain: Env.get("MSSQL_DOMAIN")}),
        server: Env.get("MSSQL_SERVER") ,
        port: Env.get("MSSQL_PORT"),
        database:  Env.get("MSSQL_DATABASE") ,
        user: Env.get("MSSQL_APP_DB_USER"),
        password: Env.get("MSSQL_APP_DB_PASSWORD"),
        options: {
            ...(Env.get("MSSQL_INSTANCE") && {instanceName: Env.get("MSSQL_INSTANCE")}),        
            trustServerCertificate: true
        }
      },
      migrations: {
        naturalSort: true,
        tableName: 'MIGRATION_VERSION_CONTROL',
        paths: Env.get('RUN_LOCALHOST_MIGRATIONS')
          ? ['./database/migrations/localhost', './database/migrations/production']
          : ['./database/migrations/production'],
        disableRollbacksInProduction: true
      },
      healthCheck: true,
      debug: (Env.get('NODE_ENV') === 'localhost'),
    }

```

As you can see, i am using some conditional props. The reason is that in another environment the mssql uses windows authentication. So i need to provide a `domain` and the `type: 'ntlm'`, but on my localhost this is not needed and the connection and migration works without problems

**Reply by solucaodev** [reply]
Hi! I rule out that there is a configuration problem, since, when running the migrations on Windows, everything works.

On Linux, the only thing that doesn't work is migrations. The connection to the database is normal and working.

See a screenshot comparing running migration:status on Windows and Ubuntu:
![image](https://user-images.githubusercontent.com/24968301/155594313-6154838a-f9d0-415a-b195-545646e19402.png)


Steps:
1 - Create any migration
2 - Using a Windows Machine, run migrations
3 - Using a Linux Machine, run migrations <---- BUG

**Reply by guntribam** [reply]
the directory structure on linux is using backslash? \

**Reply by guntribam** [reply]
sorry that i can't help more than this, but i would do this:
1 - create a slim project `npm init adonis-ts-app@latest connection-test` on linux(without any \ / mumbo jumbo, create exclusive for linux)
2 - install and configure lucid for mssql `npm i @adonisjs/lucid && node ace configure @adonisjs/lucid`
3 - create a simple migration that create a table TEST `node ace make:migration test`
4 - run the migration and check

**Reply by guntribam** [reply]
if you think the problem is in the configuration, try to check in knex github issues.

**Reply by solucaodev** [reply]
The whole problem is running the migrations using Windows, and later trying to run it again using Linux. See the image below.

I opened an ISSUE for them to verify this: https://github.com/adonisjs/lucid/issues/799

![image](https://user-images.githubusercontent.com/24968301/155628355-2c99b13f-cc66-4499-b5fc-df53e9299f19.png)

Thank you for your attention.

**Comment by solucaodev** [comment]
Analyzing via DEBUG, I found that, when using mixed environments, the bug occurs.

If you create and run migrations on Linux only, everything is fine. The same is true when you create and run migrations on Windows only. Now, when trying to run migrations that have already been run on another platform, the code fails.

See the screenshot below for an example. There is a difference in the PATHS, more precisely in the slashes from "\" to "/".
![image](https://user-images.githubusercontent.com/24968301/155600288-17b097d2-e31f-4d54-a124-9c427c9b65cb.png)
